consensus/spos/bls/subroundSignature.go

```diff
@@ -379,58 +382,84 @@ func (sr *subroundSignature) remainingTime() time.Duration {
 }
 
 func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
+
+	mutex := sync.Mutex{}
+	numMultiKeysSignaturesSent := int32(0)
+
+	sentSigForAllKeys := true
+	wg := sync.WaitGroup{}
+	wg.Add(len(sr.ConsensusGroup()))
+
+	// TODO determine first which keys are managed by current node, and try to send signatures only for those
+	for idx, pk := range sr.ConsensusGroup() {
```

Comment:
TODO should be implemented as part of this PR

suggested changes:
1. firs line of the for loop, check if the key is managed by the current node. This can be easily done with the following lines, and remove them from sendSignature method. sendSignature should just send it, not verify if it should send it or not.
```go
if !sr.IsKeyManagedByCurrentNode([]byte(pk)) {
		wg.Done()
		continue
	}
```
`wg.Done()` must be called as part of this if because we previously added to wg the total number of consensus participants, which 100% of the cases will be more than what this node manages

2. the following code should be kept here as early exit, right after the code from 1. This will prevent starting an useless goroutine.
```go
if sr.IsJobDone([]byte(pk), sr.Current()) {
	continue
}
```

3.  `numMultiKeysSignaturesSent` should be incremented based on the result of sendSignature, not inside it. Reasons: no need to manage the int32 pointer, keep the purpose of sendSignature to only send the signature

4. `sendSignature` should be renamed to `sendSignatureForManagedKey` as at the moment of calling it, we already know that it is a key managed by the current node. Also, it can return bool, no need to return error. Its signature should look like this:
```go
func (sr *subroundSignature) sendSignatureForManagedKey(idx int, pk string) bool
```

5. mutex is not really needed, we can use an atomic flag from core to keep track if the signatures were sent for all managed keys

below is how I would implement this method(+ minor changes on sendSignature)

```go
func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
	numMultiKeysSignaturesSent := uint32(0)
	signatureSentForAllManagedKeys := atomicCore.Flag{}

	wg := sync.WaitGroup{}
	wg.Add(len(sr.ConsensusGroup()))

	for idx, pk := range sr.ConsensusGroup() {
		pkBytes := []byte(pk)
		if !sr.IsKeyManagedByCurrentNode(pkBytes) {
			wg.Done()
			continue
		}

		if sr.IsJobDone(pk, sr.Current()) {
			continue
		}

		go func(managedPkIdx int, managedPk string) {
			signatureSent := sr.sendSignatureForManagedKey(managedPkIdx, managedPk)
			if signatureSent {
				atomic.AddUint32(&numMultiKeysSignaturesSent, 1)
			} else {
				signatureSentForAllManagedKeys.SetValue(false)
			}

			wg.Done()
		}(idx, pk)
	}
	wg.Wait()

	if numMultiKeysSignaturesSent > 0 {
		log.Debug("step 2: multi keys signatures have been sent", "num", numMultiKeysSignaturesSent)
	}

	return signatureSentForAllManagedKeys.IsSet()
}
```

Reply:
the wg.Add(len(sr.ConsensusGroup())) would need to be moved as well just before calling the anonymous func on a go routine and will only add 1 to wg:
wg.Add(1)

Reply:
done, I think I've covered all the suggested changes